### PR TITLE
feat: Add get user following function

### PR DESCRIPTION
**Pull request by kasugamirai**

### CODE REVIEW

File: src-tauri/src/nostr/metadata.rs [hunk 1]
```diff
@@ -469,11 +469,13 @@ pub async fn zap_event(
...
 pub async fn get_following(
-  client: &Client,
-  public_key: &PublicKey,
+  state: State<'_, Nostr>,
+  public_key: &str,
   timeout: Option<std::time::Duration>,
 ) -> Result<Vec<String>, String> {
-  let filter = Filter::new().kind(Kind::ContactList).author(*public_key);
+  let client = &state.client;
+  let public_key = PublicKey::from_str(public_key).unwrap();
```
Comment (reyamir) [inline src-tauri/src/nostr/metadata.rs:477]: I think we should have a simple validate for pubkey, because whole app will crash if pubkey is not valid here

**Comment by reyamir** [comment]
Ping me when PR is ready, I will prepare the UI for this function too. 😄

**Comment by kasugamirai** [comment]
I thought it might be hard to get the correct followers because the relay only returns a maximum of 500 events.On Jun 4, 2024, at 19:35, 雨宮蓮 ***@***.***> wrote:﻿
Ping me when PR is ready, I will prepare the UI for this function too. 😄

**Comment by reyamir** [comment]
Yes, I think there are no ways to get 100% exactly followers on Nostr anyway.

**Pull request merged**